src/tree/branch.rs

```diff
@@ -79,7 +77,8 @@ impl AsRef<[BlsScalar]> for PoseidonLevel {
     archive_attr(derive(CheckBytes))
 )]
 pub struct PoseidonBranch<const DEPTH: usize> {
-    pub(crate) path: Vec<PoseidonLevel>,
+    path: [PoseidonLevel; DEPTH],
```

Comment:
it was like this in the past, but then we had some problems in canonical regarding stack allocation. the tree serialization allocated so many branches that often we reached the OS stack limit.

do we have some assurance this won't be the case now?

Reply:
We are indeed not using `canonical` for FFI anymore. If this is too much to be on the stack we can always `Box` the branch.

Reply:
That's true, but then aren't we leaking potential problems? I'm not sure if there is a clear gain to stack allocate these, but I won't block on that since it is highly dependent on the serialization library

Reply:
If I get this right, placing it on the heap was done to skirt around a problem with canonical where it would blow the stack - I guess on deserialization. Since we're not using canonical anymore - and we are removing the feature entirely - we don't need to support this anymore.
`rkyv` our new serialization library - does zero copy deserialization, so I don't expect us blowing the stack like before. In any case it's a problem that can be solved again quickly.